run_ci_tests.sh

```diff
@@ -2,55 +2,44 @@
 
 cd $(dirname `realpath $0`)
 
-# Start the container
-export PUBLIC_KEY='ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIGzHvK2pKtSlZXP9tPYOOBb/xn0IiC9iLMS355AYUPC7'
-export DOCKER_MODS='linuxserver/mods:openssh-server-ssh-tunnel'
-
-name=openssh
-
-docker run \
-    --name $name \
-    --rm \
-    -d \
-    -p 127.0.0.1:2222:2222 \
-    -e 'USER_NAME=test-user' \
-    -e DOCKER_MODS \
-    -e PUBLIC_KEY \
-    linuxserver/openssh-server:amd64-latest
-
-function cleanup {
-    docker stop $name
-
-    # Revert modification to ~/.ssh/known_hosts
-    ssh-keygen -R "[127.0.0.1]:2222"
-
-    ssh-agent -k
-}
-trap cleanup EXIT
+./start_sshd.sh
 
 export RUSTFLAGS='--cfg=ci'
 
-cargo hack check --feature-powerset
+# Use a different target-dir since RUSTFLAGS='--cfg=ci' seems to
+# affect dependencies as well.
+#
+# Since IDEs usually do not set RUSTFLAGS='--cfg=ci', setting it
+# here would cause all the dependencies and openssh to be rebuilt.
+#
+# Thus it makes run_ci_tests.sh incrediably slow, and it also
+# affects IDEs checking, since now the IDEs also need to
+# rebuild the crate.
+CARGO_OPTS='--target-dir ci-target'
 
-cargo clippy --all-features
+cargo hack --feature-powerset check $CARGO_OPTS
+cargo clippy --all-features $CARGO_OPTS
 
 export HOSTNAME=127.0.0.1
-chmod 600 .test-key
-
-echo Waiting for sshd to be up
-while ! ssh -i .test-key -v -p 2222 -l test-user $HOSTNAME -o StrictHostKeyChecking=no whoami; do
-    sleep 3
-done
 
 echo Set up ssh agent
 eval $(ssh-agent)
 cat .test-key | ssh-add -
 
+function cleanup {
+    ssh-agent -k
+}
+trap cleanup EXIT
+
 echo Run tests
 rm -rf control-test config-file-test .ssh-connection*
 
+echo Remove all files in /tmp in the container
```

Comment:
Why? Also, it's generally preferred to quote the argument to `echo`
```suggestion
echo "Remove all files in /tmp in the container"
```

Reply:
Sftp integration tests creates files and directories in `/tmp`.

Normally, these tests would remove any files and directories  created in `/tmp`, but in case they fail, these files will be left there and interfere with the next integration test.

Reply:
Can you add a comment about that to the file?

Reply:
Sure

Reply:
Done.